app/src/sharedTest/java/org/oppia/android/app/devoptions/MarkStoriesCompletedFragmentTest.kt

```diff
@@ -0,0 +1,595 @@
+    launch<MarkStoriesCompletedTestActivity>(
+      createMarkStoriesCompletedTestActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 0)
+      performItemCheckOnStorySummaryListItem(itemPosition = 0)
+      scrollToPosition(position = 1)
+      performItemCheckOnStorySummaryListItem(itemPosition = 1)
+      scrollToPosition(position = 2)
+      performItemCheckOnStorySummaryListItem(itemPosition = 2)
+      scrollToPosition(position = 3)
+      performItemCheckOnStorySummaryListItem(itemPosition = 3)
+      scrollToPosition(position = 4)
+      performItemCheckOnStorySummaryListItem(itemPosition = 4)
+      scrollToPosition(position = 2)
+      performItemCheckOnStorySummaryListItem(itemPosition = 2)
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.mark_stories_completed_all_check_box)).check(matches(not(isChecked())))
+    }
+  }
+
+  @Test
+  fun testMarkStoriesCompletedFragment_ratiosFirstStoryIsCompleted_isCheckedAndDisabled() {
+    markRatiosFirstStoryCompleted()
+    launch<MarkStoriesCompletedTestActivity>(
+      createMarkStoriesCompletedTestActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 3)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.mark_stories_completed_recycler_view,
+          position = 3,
+          targetViewId = R.id.mark_stories_completed_story_check_box
+        )
+      ).check(matches(isChecked())).check(matches(not(isEnabled())))
+    }
+  }
+
+  @Test
+  fun testMarkStoriesCompletedFragment_ratiosFirstStoryIsCompleted_configChange_isCheckedAndDisabled() { // ktlint-disable max-line-length
+    markRatiosFirstStoryCompleted()
+    launch<MarkStoriesCompletedTestActivity>(
+      createMarkStoriesCompletedTestActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(isRoot()).perform(orientationLandscape())
+      scrollToPosition(position = 3)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.mark_stories_completed_recycler_view,
+          position = 3,
+          targetViewId = R.id.mark_stories_completed_story_check_box
+        )
+      ).check(matches(isChecked())).check(matches(not(isEnabled())))
+    }
+  }
+
+  @Test
+  fun testMarkStoriesCompletedFragment_clickMarkCompleted_activityFinishes() {
+    activityTestRule.launchActivity(createMarkStoriesCompletedTestActivityIntent(internalProfileId))
+    testCoroutineDispatchers.runCurrent()
+    onView(withId(R.id.mark_stories_completed_mark_completed_text_view)).perform(click())
+    assertThat(activityTestRule.activity.isFinishing).isTrue()
+  }
+
+  @Test
+  fun testMarkStoriesCompletedFragment_configChange_clickMarkCompleted_activityFinishes() {
```

Comment:
Make sure you add `land` instead of `configChange` if you're testing for landscape.

Reply:
Either one is fine I think.

Reply:
Actually, `land` is just an alternative for `configChange`. I used this to fit the test name in the 100 char limit. This was suggested in one of my earlier PRs https://github.com/oppia/oppia-android/pull/3445#discussion_r666338923. I think there should not be an issue with this.

Reply:
Renamed.